corehq/apps/integration/payments/services.py

```diff
@@ -167,3 +169,20 @@ def _payment_is_verified(case_data: dict):
 
 def _payment_already_submitted(case_data: dict):
     return case_data.get(PaymentProperties.PAYMENT_SUBMITTED) == 'True'
+
+
+def get_payment_batch_numbers_for_domain(domain):
+    case_ids = (
+        CaseSearchES()
+        .domain(domain)
+        .case_type(MOMO_PAYMENT_CASE_TYPE)
+        .values_list("_id", flat=True)
+    )
+
+    batch_numbers = set()
+    for case_id_chunk in chunked(case_ids, CHUNK_SIZE):
+        cases = CommCareCase.objects.get_cases(case_ids=list(case_id_chunk), domain=domain)
+        chunk_batch_numbers = set([case.case_json.get(PaymentProperties.BATCH_NUMBER) for case in cases])
+        batch_numbers.update(chunk_batch_numbers)
+
+    return sorted([batch_number for batch_number in batch_numbers if batch_number])
```

Comment:
nit: is sorting needed? I'd assume that someone would only type and search?
Just thinking about making this as efficient as possible.

Reply:
Not strictly needed, no. The user is able to type and search. I'll update.